modules/average_month.py

```diff
@@ -0,0 +1,93 @@
+"""
+This module generates a report which displays average highest and lowest temperature
+as well as average mean humidity for a given month
+"""
+from constants import WEATHER_FILES_DIR, months_list
+from modules.utils import (
+    get_highest_temperature,
+    get_lowest_temperature,
+    get_mean_humidity,
+    read_data,
+)
+
+
+def calculate_averages(pattern, path):
+    """
+    Calculates average of highest_temperature, lowest_temperature and mean_humidity
+    from files with names that match a pattern in given path
+    1. avg highest temperature
+    2. avg lowest temperature
+    3. avg mean humidity
+    Args:
+        pattern(str): a string which contains a year/month e.g '2006/5', '2007/3'
+        path(str): a value containing path to weather files e.g: 'weatherfiles/'
+    Returns:
+        (list or None): list of length 3, averages where
+                        averages[0] = avg_highest_temperature
+                        averages[1] = avg_lowest_temperature
+                        averages[2] = avg_mean_humidity
+                        Or
+                        None for failure
+    """
+    sum_highest_temperature = 0
+    sum_lowest_temperature = 0
+    sum_mean_humidity = 0
+    lines = []
+    for lines in read_data(pattern, path):
+        if lines:
```

Comment:
If lines is None would the loop continue? Do we need this check here?

Reply:
No, the loop would not continue. we don't need this check here. removed it